tools/llzk-tblgen/CommonCAPIGen.h

```diff
@@ -0,0 +1,661 @@
+/// @endcode
+///
+/// Example output:
+/// - ExtraMethod { returnType="unsigned", methodName="getWidth", isConst=true, hasParameters=false
+/// }
+/// - ExtraMethod { returnType="bool", methodName="isSignless", isConst=true, hasParameters=false }
+///
+/// Note: Methods with parameters are detected but currently skipped during code generation.
+llvm::SmallVector<ExtraMethod> parseExtraMethods(mlir::StringRef extraDecl);
+
+/// @brief Check if a C++ type matches an MLIR type pattern
+/// @param cppType The C++ type to check
+/// @param typeName The MLIR type name to match against
+/// @return true if the C++ type matches the MLIR type
+bool matchesMLIRClass(mlir::StringRef cppType, mlir::StringRef typeName);
+
+/// @brief Convert C++ type to MLIR C API type
+/// @param cppType The C++ type to convert
+/// @return The corresponding MLIR C API type if convertible, std::nullopt otherwise
+std::optional<std::string> tryCppTypeToCapiType(mlir::StringRef cppType);
+
+/// @brief Map C++ type to corresponding C API type
+/// @param cppType The C++ type to map
+/// @return The corresponding C API type string
+///
+/// @note This function should not be called directly for ArrayRef types.
+/// Use extractArrayRefElementType() first and then use this on the result.
+std::string mapCppTypeToCapiType(mlir::StringRef cppType);
+
+/// @brief Base class for C API generators
+struct Generator {
+  Generator(std::string_view recordKind, llvm::raw_ostream &outputStream)
+      : kind(recordKind), os(outputStream), dialectNameCapitalized(toPascalCase(DialectName)) {}
+  virtual ~Generator() = default;
+
+  /// @brief Set the dialect and class name for code generation
+  /// @param d Pointer to the dialect definition
+  /// @param cppClassName The C++ class name of the entity being generated
+  virtual void
+  setDialectAndClassName(const mlir::tblgen::Dialect *d, mlir::StringRef cppClassName) {
+    this->dialect = d;
+    this->className = cppClassName;
+  }
+
+  /// @brief Generate code for extra methods from an `extraClassDeclaration`
+  /// @param extraDecl The extra class declaration string
+  virtual void genExtraMethods(mlir::StringRef extraDecl) const {
+    if (extraDecl.empty()) {
+      return;
+    }
+    for (const ExtraMethod &method : parseExtraMethods(extraDecl)) {
+      genExtraMethod(method);
+    }
+  }
+
+  /// @brief Generate code for an extra method
+  /// @param method The extra method to generate code for
+  virtual void genExtraMethod(const ExtraMethod &method) const = 0;
+
+protected:
+  std::string kind;
+  llvm::raw_ostream &os;
+  std::string dialectNameCapitalized;
+  const mlir::tblgen::Dialect *dialect;
+  mlir::StringRef className;
+};
+
+/// @brief Generator for common C header file elements
+struct HeaderGenerator : public Generator {
+  using Generator::Generator;
+  virtual ~HeaderGenerator() = default;
+
+  virtual void genPrologue() const {
+    os << R"(
+#include "llzk-c/Builder.h"
+#include "mlir-c/IR.h"
```

Comment:
```suggestion
#include <llzk-c/Builder.h>
#include <mlir-c/IR.h>
```
eh?

Reply:
We've used `<>` for `mlir-c` elsewhere but not `llzk-c`